packages/react/src/vertex4.tsx

```diff
@@ -0,0 +1,322 @@
+
+    let labelShapeWidth = 0;
+    if (text !== "") {
+        labelShapeWidth = labelWidth + (textPadding * 2) + (textboxStrokeWidth * 2);
+    }
+    let fullAnnotationWidth = labelShapeWidth + annotationGutter;
+    const textOffsetX = fullAnnotationWidth - (labelShapeWidth / 2);
+    
+    const textShapeHeight = textHeight + (annotationGutter*2) + (textboxStrokeWidth * 2);
+    const annoWidthArr = React.useMemo(() => {
+        return annotations.map((anno, i) => {
+            return Utility.textSize(anno.imageChar, anno.imageFontFamily, anno.height, false).width;
+        });
+    }, [annotations]);
+    const annotationArr = React.useMemo(() => {
+        const annotationArr = [];
+        let _labelAnnoOffsetX = fullAnnotationWidth;
+        annotations.forEach((anno, i) => {
+            const annoText = anno.imageChar;
+            const annoTextHeight = anno.height ?? textShapeHeight;
+            _labelAnnoOffsetX += annoWidthArr[i] + annotationGutter;
+            const annoOffsetX = _labelAnnoOffsetX - (annoWidthArr[i] / 2);
+            annotationArr.push(
+                <g key={i} class="vertex3-anno" data-anno={JSON.stringify(anno)} transform={`translate(${annoOffsetX} ${annoOffsetY})`}>
+                    <Icon
+                        {...anno}
+                        shape="rectangle"
+                        width={annoWidthArr[i]}
+                        height={annoTextHeight}
+                        imageChar={annoText}
+                        imageFontFamily={anno.imageFontFamily}
+                        cornerRadius={cornerRadius}
+                        strokeWidth={0}
+                    />
+                </g>
+            );
+        });
+        return annotationArr;
+    }, [fullAnnotationWidth, annotations, textShapeHeight, annoWidthArr, annotationGutter, cornerRadius]);
+    fullAnnotationWidth += annoWidthArr.reduce((r, n)=>r+n, 0);
+    
+    if (annotations.length > 0) {
+        fullAnnotationWidth += annotationGutter * (annotations.length - 1);
+    }
+    const iconAnnotationArr = React.useMemo(() => {
+        const iconAnnotationArr = [];
+        iconAnnotations.forEach((anno, i) => {
+            const x = anno.shapeOffsetX;
+            const y = anno.shapeOffsetY;
+            iconAnnotationArr.push(
+                <g key={i} class="vertex3-iconAnno" data-anno={JSON.stringify(anno)} transform={`translate(${x} ${y})`}>
+                    <Icon
+                        {...anno}
+                        shape={anno.shape ?? "square"}
+                        imageChar={anno.imageChar}
+                        imageFontFamily={anno.imageFontFamily}
+                        cornerRadius={cornerRadius}
+                        stroke={anno.stroke}
+                        strokeWidth={anno.strokeWidth}
+                    />
+                </g>
+            );
+        });
+        return iconAnnotationArr;
+    }, [iconAnnotations, cornerRadius]);
```

Comment:
This will only update if the iconAnnotations array reference changes, not if the content of the iconAnnotations array changes.